core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2222,6 +2223,9 @@ private[spark] object Utils extends Logging {
         isBindCollision(e.getCause)
       case e: MultiException =>
         e.getThrowables.asScala.exists(isBindCollision)
+      case e: NativeIoException =>
+        e.getMessage().startsWith("bind() failed: ") ||
```

Comment:
Seems OK. Do you need to check for a null message though, like above? In fact, the condition above for `BindException` could be `e.getMessage != null || isBindCollision(e.getCause)`